src/System.Drawing.Primitives/tests/SizeTests.netcoreapp.cs

```diff
@@ -0,0 +1,132 @@
+        }
+
+
+        [Theory]
+        [InlineData(1000, 2400.933f)]
+        [InlineData(int.MaxValue, 0.0f)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MinValue, 0.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        public void MultiplicationTestSizeFloat(int value1, float value2)
+        {
+            Size sz1 = new Size(value1, value1);
+            SizeF mulExpected;
+
+            unchecked
+            {
+                mulExpected = new SizeF(value1 * value2, value1 * value2);
+            }
+
+            Assert.Equal(mulExpected, sz1 * value2);
+            Assert.Equal(mulExpected, value2 * sz1);
+        }
+
+        [Fact]
+        public void DivideByZeroChecks()
+        {
+            Size size = new Size(100, 100);
+            Assert.Throws<DivideByZeroException>(() => size / 0);
+
+            float invDiv = 1 / 0.0f;
+            SizeF expectedSizeF = new SizeF(float.PositiveInfinity, float.PositiveInfinity);
+            Assert.Equal(expectedSizeF, size * invDiv);
+        }
+
+        [Theory]
+        [InlineData(0, 1)]
+        [InlineData(1, 1)]
+        [InlineData(-1, 1)]
+        [InlineData(1, -1)]
+        [InlineData(-1, -1)]
+        [InlineData(int.MaxValue, int.MaxValue)]
+        [InlineData(int.MaxValue, int.MinValue)]
+        [InlineData(int.MinValue, int.MaxValue)]
+        [InlineData(int.MinValue, int.MinValue)]
+        [InlineData(int.MaxValue, 1)]
+        [InlineData(int.MinValue, 1)]
+        [InlineData(int.MaxValue, -1)]
+        public void DivideTestSizeInt(int value1, int value2)
+        {
+            Size size = new Size(value1, value1);
+            Size expected;
+
+            int invDiv = 1 / value2;
+            expected = new Size(value1 * invDiv, value1 * invDiv);
+
+            Assert.Equal(expected, size / value2);
+        }
+
+        [Theory]
+        [InlineData(0, 1.0f)]
+        [InlineData(1, 1.0f)]
+        [InlineData(-1, 1.0f)]
+        [InlineData(1, -1.0f)]
+        [InlineData(-1, -1.0f)]
+        [InlineData(int.MaxValue, float.MaxValue)]
+        [InlineData(int.MaxValue, float.MinValue)]
+        [InlineData(int.MinValue, float.MaxValue)]
+        [InlineData(int.MinValue, float.MinValue)]
+        [InlineData(int.MaxValue, 1.0f)]
+        [InlineData(int.MinValue, 1.0f)]
+        [InlineData(int.MaxValue, -1.0f)]
+        [InlineData(int.MinValue, -1.0f)]
+        public void DivideTestSizeFloat(int value1, float value2)
+        {
+            SizeF size = new SizeF(value1, value1);
+            SizeF expected;
+
+            float invDiv = 1.0f / value2;
+
+            expected = new SizeF(value1 * invDiv, value1 * invDiv);
```

Comment:
Bringing back the comment that I had mentioned above previously... In case of division for combinations (Size, float) and (SizeF, float), not doing reciprocal multiplication in test case is not giving results identical to operator /. E.g.

```
     System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(value1: 2147483647, value2: 3.402823E+38) [FAIL]
        Assert.Equal() Failure
        Expected: {Width=6.310888E-30, Height=6.310888E-30}
        Actual:   {Width=6.310887E-30, Height=6.310887E-30}
        Stack Trace:
           D:\WinCPP\CoreFx\src\System.Drawing.Primitives\tests\SizeTests.netcoreapp.cs(125,0): at System.Drawing.PrimitivesTests.SizeTests.DivideTestSizeFloat(Int32 value1, Single value2)

     System.Drawing.PrimitivesTest.SizeFTests.DivideTestSizeFloat(value1: -3.402823E+38, value2: -3.402823E+38) [FAIL]
        Assert.Equal() Failure
        Expected: {Width=1, Height=1}
        Actual:   {Width=0.9999999, Height=0.9999999}
        Stack Trace:
           D:\WinCPP\CoreFx\src\System.Drawing.Primitives\tests\SizeFTests.netcoreapp.cs(64,0): at System.Drawing.PrimitivesTest.SizeFTests.DivideTestSizeFloat(Single value1, Single value2)
```

I thought that the `operator /` was just a syntactic sugar that worked on _whole_ in a way identical to how it would work on _parts_... But by using different approaches inside operator and in test case, are we ending up validating floating point division...? The failures are for these combinations,
```
operator / in Size
        [InlineData(int.MaxValue, float.MaxValue)]
        [InlineData(int.MaxValue, float.MinValue)]
        [InlineData(int.MinValue, float.MaxValue)]
        [InlineData(int.MinValue, float.MinValue)]

operator / in SizeF
        [InlineData(float.MaxValue, float.MaxValue)]
        [InlineData(float.MaxValue, float.MinValue)]
        [InlineData(float.MinValue, float.MaxValue)]
        [InlineData(float.MinValue, float.MinValue)]